audio/openal-soft-devel/Portfile

```
@@ -0,0 +1,110 @@
+                        3D environment. Distance attenuation, doppler shift, \
+                        and directional sound emitters are among the features \
+                        handled by the API. More advanced effects, including \
+                        air absorption, occlusion, and environmental reverb, \
+                        are available through the EFX extension. It also \
+                        facilitates streaming audio, multi-channel buffers, \
+                        and audio capture.
+
+homepage                https://openal-soft.org
+
+patchfiles-append       0001-coreaudio.cpp-restore-fallbacks-for-10.5.patch
+
+depends_build-append    path:bin/pkg-config:pkgconfig
+
+depends_lib-append      port:libmysofa \
+                        port:libsndfile \
+                        port:zlib
+
+compiler.cxx_standard           2017
+
+compiler.thread_local_storage   yes
+
+# error: inline declaration of 'configValue<...>' follows non-inline definition
+# https://github.com/kcat/openal-soft/issues/703
+compiler.blacklist-append {clang < 900}
+
+configure.args-append   -DALSOFT_EXAMPLES=OFF \
+                        -DALSOFT_UTILS=ON \
+                        -DALSOFT_NO_CONFIG_UTIL=ON \
+                        -DALSOFT_REQUIRE_JACK=OFF \
+                        -DALSOFT_BACKEND_JACK=OFF \
+                        -DALSOFT_REQUIRE_PORTAUDIO=OFF \
+                        -DALSOFT_BACKEND_PORTAUDIO=OFF \
+                        -DALSOFT_REQUIRE_PULSEAUDIO=OFF \
+                        -DALSOFT_BACKEND_PULSEAUDIO=OFF \
+                        -DALSOFT_REQUIRE_SDL2=OFF \
+                        -DALSOFT_BACKEND_SDL2=OFF
+
+platform darwin {
+    # https://github.com/kcat/openal-soft/issues/1066
+    if {${os.major} <= 16} {
+        if {[string match *clang* ${configure.compiler}]} {
+            configure.cxxflags-append \
+                        -fno-aligned-allocation
+        }
+    }
```

It is absolutely not safe to pass `-fno-aligned-allocation` just to silence an error, because it will result in incorrect behaviour within openal-soft.

What's needed here is to provide a set of working aligned allocator functions (based on `posix_memalign` or something, like openal-soft used to use) with the right names, and pass `-faligned-allocation` to silence the error.

---

I can drop this chunk altogether, the port will just not build on affected systems, but for a -devel subport this is not a big concern. My idea was to have it exactly to be able to test stuff without affecting the main `openal-soft` port.

How does it work with gcc now, do you know? If it was `aligned_alloc`, gcc build would fail without a fallback, but it does not fail (no flags or hacks needed).

---

I assume it works with gcc 14 because libgcc provides an implementation of the aligned allocators, but libc++ does not on macOS <10.13.

Just a note for anyone that wants to look at this further: `posix_memalign` is only supported on >=10.6

---

> Just a note for anyone that wants to look at this further: posix_memalign is only supported on >=10.6

Yes, but `legacysupport` supplies it for < 10.6, AFAIK.